File: examples/eop/file_provider.py
# /// script
# dependencies = ["brahe"]
# ///
"""
Using FileEOPProvider with downloaded EOP data.

Demonstrates:
- Downloading EOP files from IERS
- Loading EOP from file
- Setting global file-based provider
- Interpolation and extrapolation options
"""

import brahe as bh
import tempfile
import os

if __name__ == "__main__":
    # Create temporary directory for EOP data
    temp_dir = tempfile.mkdtemp()
    eop_file = os.path.join(temp_dir, "finals.all.iau2000.txt")

    # Download latest EOP file
    print("Downloading EOP file...")
    bh.download_standard_eop_file(eop_file)
    print(f"Downloaded to {eop_file}")

    # Load from file with interpolation
    provider = bh.FileEOPProvider.from_file(
        eop_file, interpolate=True, extrapolate="Hold"
    )
    bh.set_global_eop_provider_from_file_provider(provider)

    print("File EOP provider initialized")
    print("Use case: Production applications with current EOP data")

    # Cleanup
    os.remove(eop_file)
    os.rmdir(temp_dir)

File: examples/eop/file_provider.rs
//! Using FileEOPProvider with downloaded EOP data.
//!
//! Demonstrates:
//! - Downloading EOP files from IERS
//! - Loading EOP from file
//! - Setting global file-based provider
//! - Interpolation and extrapolation options

use brahe::eop::*;
use std::path::Path;
use std::fs;
use std::env;

fn main() {
    // Create temporary directory for EOP data
    let temp_dir = env::temp_dir();
    let eop_file = temp_dir.join("finals.all.iau2000.txt");

    // Download latest EOP file
    println!("Downloading EOP file...");
    download_standard_eop_file(eop_file.to_str().unwrap()).unwrap();
    println!("Downloaded to {:?}", eop_file);

    // Load from file with interpolation
    let provider = FileEOPProvider::from_file(
        &eop_file,  // Use &Path directly
        true,  // interpolate
        EOPExtrapolation::Hold
    ).unwrap();
    set_global_eop_provider(provider);

    println!("File EOP provider initialized");
    println!("Use case: Production applications with current EOP data");

    // Cleanup
    fs::remove_file(eop_file).ok();
}
